tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -2328,17 +2329,44 @@ public function updateSharedResource(string $user, array $rows):ResponseInterfac
 	}
 
 	/**
-	 * @When /^user "([^"]*)" expires the last share$/
+	 * @When user :user expires the last share of resource :resource inside of the space :spaceName
 	 *
 	 * @param string $user
+	 * @param string $resource
+	 * @param string $spaceName
 	 *
 	 * @return void
+	 * @throws GuzzleException|JsonException
 	 */
-	public function userExpiresLastResourceShare(string $user): void {
+	public function userExpiresTheLastShareOfResourceInsideOfTheSpace(string $user, string $resource, string $spaceName): void {
 		$dateTime = new DateTime('yesterday');
 		$rows['expireDate'] = $dateTime->format('Y-m-d\\TH:i:sP');
-		$rows['permissions'] = (string) $this->featureContext->getLastCreatedPublicShare()->permissions;
-		$this->featureContext->setResponse($this->updateSharedResource($user, $rows));
+		if ($this->featureContext->isUsingSharingNG()) {
+			if (!\in_array($spaceName, ['Personal', 'Shares'])) {
+				$space = $this->getSpaceByName($user, $spaceName);
+				$itemId = $this->getResourceId($user, $spaceName, $resource);
+			} else {
+				$space = $this->getCreatedSpace($spaceName);
+				$itemId = $space['fileId'];
+			}
```

Comment:
will this be possible? because this step should only be for resource
```suggestion
			
			$space = $this->getSpaceByName($user, $spaceName);
			$itemId = $this->getResourceId($user, $spaceName, $resource);
```

Reply:
Yeah. it works.